BulkyWeb/Controllers/CategoryController.cs

```diff
@@ -16,5 +16,83 @@ public IActionResult Index()
             List<Category> objCategoryList = _db.Categories.ToList();
             return View(objCategoryList);
         }
+
+        public IActionResult Create()
+        {
+            return View();
+        }
+        [HttpPost]
+        public IActionResult Create(Category obj)
+        {
+            if (obj.Name == obj.DisplayOrder.ToString())
+            {
+                ModelState.AddModelError("name", "The DisplayOrder cannot exactly match the Name.");
+            }
+            if (ModelState.IsValid)
+            {
+                _db.Categories.Add(obj);
+                _db.SaveChanges();
+                TempData["success"] = "Category created successfully";
+                return RedirectToAction("Index");
+            }
+            return View();
+        }
+
+        public IActionResult Edit(int? id)
+        {
+            if(id==null || id == 0)
+            {
+                return NotFound();
+            }
+            Category? categoryFromDb = _db.Categories.Find(id);
```

Comment:
Same thing here. Look more into the ? in c# and understand where and when to use it

Reply:
Also look into Null-coalescing operator and
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator

Reply:
I will.